### PR TITLE
[lodash] Improve isWeak{Set,Map} function

### PR DESCRIPTION
**Pull request by falsandtru**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://github.com/Microsoft/TypeScript/pull/19756>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/lodash/index.d.ts
Comment (DanielRosenwasser) [inline types/lodash/index.d.ts]: Would it work to just make this something like `isWeakMap(value?: any): value is WeakMap<object, any>`?

Comment (falsandtru) [reply types/lodash/index.d.ts]: Done!

File: types/lodash/index.d.ts
Comment (thorn0) [inline types/lodash/index.d.ts]: Just like `isWeakMap`, this method should be made non-generic. Otherwise it'd be a disguised type assertion (see [Common Mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes)).

**Review by thorn0 — CHANGES_REQUESTED** [review]

File: types/lodash/index.d.ts
Comment (falsandtru) [reply types/lodash/index.d.ts]: It is needed for compatibility with a fix of TypeScript. See Microsoft/TypeScript#19756. It is blocked by lodash.

**Comment by RyanCavanaugh** [comment]
@falsandtru Please address comments from the code reviewers.

**Comment by falsandtru** [comment]
I'm not sure what should I do. You mean `isWeakSet(value?: any): value is WeakSet<object>;`?

**Comment by falsandtru** [comment]
Anyways, I done.

**Comment by RyanCavanaugh** [comment]
@thorn0 - Thanks for your review of this PR! Can you please look at the new code and update your review status if appropriate?

**Comment by thorn0** [comment]
Looks good (even though I probably don't understand fully the compatibility issue with Microsoft/TypeScript#19756).

**Review by thorn0 — APPROVED** [review]

**Comment by RyanCavanaugh** [comment]
Approved by a listed owner. PR ready to merge pending express review by a maintainer.

**Comment by falsandtru** [comment]
@DanielRosenwasser @RyanCavanaugh Can you merge?

**Pull request merged**
